src/mixin/ApproveAndBridge.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8;
+
+import {IApproveAndBridge} from "../interface/IApproveAndBridge.sol";
+import {IERC20} from "../vendored/IERC20.sol";
+import {SafeERC20} from "../vendored/SafeERC20.sol";
+
+abstract contract ApproveAndBridge is IApproveAndBridge {
+    using SafeERC20 for IERC20;
+
+    error MinAmountNotMet();
+
+    /// @dev Address used to represent the native token
+    address public constant NATIVE_TOKEN_ADDRESS = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+
+    /// @dev This function isn't intended to be called directly, it should be
+    /// delegatecalled instead.
+    function approveAndBridge(IERC20 token, uint256 minAmount, bytes calldata data) external {
+        // get the balance of the token
+        uint256 balance =
+            address(token) == NATIVE_TOKEN_ADDRESS ? address(this).balance : token.balanceOf(address(this));
+
+        // check if the balance is greater than the minAmount
+        if (balance < minAmount) revert MinAmountNotMet();
```

Comment:
I don't think is an issue, but probably we should add in the `@dev` documentation that `minAmount` should be an amount not too small if the sell amount is big (basically, the order should have reasonable slippage tolerances). A slippage tolerance of 99.99% could make it almost free for anyone to "consume the hook" making the user's bridge to fail (by a donation of the `minAmount`).